Android-Sample/README.md

```diff
@@ -16,217 +20,108 @@ The sample app only subscribes to the test stream. It does not subscribe to othe
-   must be a different session than the one you will use to share audio-video streams between
-   clients. Do not publish more than one stream to the test session.
+    Make sure you've imported the `NetworkQualityTest` class into your project.
 
-   The app requires that each session uses the routed media mode -- one that uses
-   the [OpenTok Media Router](https://tokbox.com/developer/guides/create-session/#media-mode).
-   A routed session is required to get statistics for the stream published by the local client.
+    ```java
+    import com.opentok.qualitystats.sample.NetworkQualityTest;
+    ```
 
-   You can get your API key as well as a test session ID and token at the
-   [OpenTok dashboard](https://dashboard.tokbox.com/). However, in a shipping application, use
-   one of the [OpenTok server SDKs](https://tokbox.com/developer/sdks/server/) to generate a
-   session ID and token.
+2. **Instantiate the Class**
 
-4. Debug the project on a supported device.
+    Create an instance of `NetworkQualityTest` by providing the required parameters: the current activity, a configuration object, and a callback listener for receiving the test results.
 
-   For a list of supported devices, see the "Developer and client requirements"
-   on [this page](https://tokbox.com/developer/sdks/android/).
+    ```java
+    NetworkQualityTest networkQualityTest = new NetworkQualityTest(
+        this,  // Current activity
+        networkQualityTestConfig,  // Configuration object
+        networkQualityTestCallbackListener  // Callback listener
+    );
+    ```
 
- The app uses a test stream and session to determine the client's ability to publish a stream
- that has audio and video. At the end of the test it reports one of the following:
+3. **NetworkQualityTestConfig**
 
-   * You're all set -- your client can publish an audio-video stream.
+    The `NetworkQualityTestConfig` class is used to configure the parameters of the network quality test. It provides the following constructor:
 
-   * Voice-only -- Your bandwidth is too low for video.
+    You can configure the following parameters:
+    
+    - `sessionId`: The session ID of your OpenTok session.
+    - `apiKey`: Your OpenTok API key.
+    - `token`: The token for connecting to the session.
+    - `resolution`: The camera capture resolution for the publisher. Defaults to `HIGH` if not specified.
+    - `testDurationSec`: The duration of the test in seconds. Defaults to `30 sec` if not specified.
 
-   * You can't connect successfully -- Your bandwidth is too low for audio or video.
+4. **Handle Callbacks**
 
-## Understanding the code
+    Once the test is running, you'll receive callbacks in the provided listener methods:
+    
+    - `onQualityTestResults`: Receive the final test results and the recommended setting based on quality thresholds.
+    - `onQualityTestStatsUpdate`: Receive real-time statistics updates during the test.
+    - `onError`: Handle errors that occur during the test.
```

Comment:
Please provide more details on the API:

* `onQualityTestResults()` -- What are the possible values of `recommendedSetting`, and what do they mean?
* `onQualityTestStatsUpdate()` -- What are the properties of `CallbackQualityStats`, and what do they mean?
* `onError()` -- What are the possible errors, and what do they mean?